magefiles/helm.go

```diff
@@ -0,0 +1,110 @@
+//go:build mage_helm
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/magefile/mage/sh"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v3"
+)
+
+const chartFile = "./helm/trivy/Chart.yaml"
+
+func main() {
+	trivyVersion, err := version()
+	if err != nil {
+		log.Fatalf("could not determine Trivy version: %v", err)
+	}
+	input, err := os.ReadFile(chartFile)
+	if err != nil {
+		log.Fatalf("could not find helm chart %s: %v", chartFile, err)
+	}
+	jsonData := map[string]interface{}{}
+	if err := yaml.Unmarshal(input, &jsonData); err != nil {
+		log.Fatalf("could not unmarshal helm chart %s: %v", chartFile, err)
+	}
+	currentAppVersion, ok := jsonData["appVersion"].(string)
+	if !ok {
+		log.Fatalf("could not determine current app version")
+	}
+	currentHelmVersion, ok := jsonData["version"].(string)
+	if !ok {
+		log.Fatalf("could not determine current helm version")
+	}
+	newHelmVersion := newHelmVersion(currentHelmVersion, currentAppVersion, trivyVersion)
+
+	log.Printf("Current helm version %q with Trivy %q will bump up %q with Trivy %q",
+		currentHelmVersion, currentAppVersion, newHelmVersion, trivyVersion)
+
+	newBranch := fmt.Sprintf("ci/helm-chart/bump-trivy-to-%s", trivyVersion)
+	title := fmt.Sprintf("ci(helm): bump Trivy version to %s", trivyVersion)
+	description := fmt.Sprintf("This PR bumps Trivy up to the %s version for the Helm chart.", trivyVersion)
+
+	cmds := [][]string{
+		[]string{"sed", "-i", "-e", fmt.Sprintf("s/appVersion: %s/appVersion: %s/g", currentAppVersion, trivyVersion), chartFile},
+		[]string{"sed", "-i", "-e", fmt.Sprintf("s/version: %s/version: %s/g", currentHelmVersion, trivyVersion), chartFile},
```

Comment:
It seems like overwriting the versions in `jsonData` and writing this yaml would be more reliable.
wdyt?

Reply:
I tried several ways to overwriting `Chart.yaml`. 
Martshaling/Unmarshaling doesn't allow to keep order of unknown fields.
If we create a custom struct we can miss something in the future.

using a specific library (https://pkg.go.dev/helm.sh/helm/v3/pkg/chart) doesn't allow to save only `Chart.yaml`, it creates all helm files.

so `sed` is the simplest way to update the yaml file.
wdyt?

Reply:
Thanks for checking this out.
Then let's use `sed`, I don't mind 👍